### PR TITLE
Share a case core

### PR DESCRIPTION
**Pull request by makowalski**
### JIRA link (if applicable) ###
[RIA-723](https://tools.hmcts.net/jira/browse/RIA-723)

### Change description ###
partially reverting changes for Direction object

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: charts/ia-case-api/values.yaml [hunk 1]
```diff
@@ -7,6 +7,7 @@ java:
     POSTGRES_HOST: ia-case-api-postgres-db-{{ .Values.global.environment }}.postgres.database.azure.com
     POSTGRES_USERNAME: ia_case_api@ia-case-api-postgres-db-{{ .Values.global.environment }}
     POSTGRES_CONNECTION_OPTIONS: "?sslmode=require"
+    PROF_REF_DATA_URL: "http://rd-professional-api-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal"
```
Comment (raghera) [inline charts/ia-case-api/values.yaml]: If we do this ... in theory we would not need the values in the vault as it sets the env correctly from global vars.

File: src/functionalTest/java/uk/gov/hmcts/reform/iacaseapi/CcdScenarioRunnerTest.java [hunk 2]
```diff
@@ -134,6 +134,8 @@ public void scenarios_should_behave_as_specified() throws IOException {
             final String requestUri = MapValueExtractor.extract(scenario, "request.uri");
             final int expectedStatus = MapValueExtractor.extractOrDefault(scenario, "expectation.status", 200);
 
+            System.out.println();
```
Comment (raghera) [inline src/functionalTest/java/uk/gov/hmcts/reform/iacaseapi/CcdScenarioRunnerTest.java]: can probably get rid of this

File: src/functionalTest/java/uk/gov/hmcts/reform/iacaseapi/ShareACaseScenarioTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
+package uk.gov.hmcts.reform.iacaseapi;
```
Comment (raghera) [inline src/functionalTest/java/uk/gov/hmcts/reform/iacaseapi/ShareACaseScenarioTest.java]: This file is not required.

File: src/functionalTest/java/uk/gov/hmcts/reform/iacaseapi/CcdScenarioRunnerTest.java (hunk 2 above)
Comment (makowalski) [reply src/functionalTest/java/uk/gov/hmcts/reform/iacaseapi/CcdScenarioRunnerTest.java]: :+1:

File: src/functionalTest/java/uk/gov/hmcts/reform/iacaseapi/ShareACaseScenarioTest.java (hunk 3 above)
Comment (makowalski) [reply src/functionalTest/java/uk/gov/hmcts/reform/iacaseapi/ShareACaseScenarioTest.java]: :+1:

File: charts/ia-case-api/values.yaml (hunk 1 above)
Comment (makowalski) [reply charts/ia-case-api/values.yaml]: as discussed, there is a need to refactor this and other urls which shouldn't be stored in Vault

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/ShareACaseUserListPreparer.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+        AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+
+        DynamicList dynamicList;
+
+        final List<Value> values =
+            usersResponse.getProfessionalUsers()
+                .stream()
+                .filter(professionalUser ->
+                    professionalUser.getIdamStatus().equalsIgnoreCase("ACTIVE"))
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/ShareACaseUserListPreparer.java]: In future we can use the parameter in PRD API that allows a filtered response.  This is good for the core version

Comment (makowalski) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/ShareACaseUserListPreparer.java]: :+1:

**Pull request merged**